### PR TITLE
convert Button.vue to typescript

### PR DESCRIPTION
**Pull request by ToBinio**
Convert `Button.vue` to use Typescript and change how color classes are set

### CODE REVIEW

File: lib/components/base/Button.vue [hunk 1]
```diff
@@ -1,50 +1,45 @@
...
+      | 'highlight'
+    iconOnly: boolean
+    large: boolean
+    outline: boolean
+    transparent: boolean
+    hoverFilled: boolean
+    hoverFilledOnly: boolean
+  }>(),
+  {
+    link: undefined,
```
Comment (brawaru) [inline lib/components/base/Button.vue:31]: Wouldn't it be `undefined` anyway if you skipped it?

File: lib/components/base/Button.vue [hunk 2]
```diff
@@ -58,21 +53,12 @@ const accentedButton = computed(() =>
...
-      'btn-green': color === 'green',
-      'btn-blue': color === 'blue',
-      'btn-purple': color === 'purple',
-      'btn-gray': color === 'gray',
       'btn-transparent': transparent,
       'btn-hover-filled': hoverFilled,
       'btn-hover-filled-only': hoverFilledOnly,
       'btn-outline': outline,
       'color-accent-contrast': accentedButton,
+      ['btn-' + props.color]: true,
```
Comment (brawaru) [inline lib/components/base/Button.vue:61]:
```
      [`btn-${color}`]: true,
```

Use string literal. Also I believe direct prop access would work here in the template.

File: lib/components/base/Button.vue [hunk 3]
```diff
@@ -87,21 +73,12 @@ const accentedButton = computed(() =>
...
```
Comment (brawaru) [inline lib/components/base/Button.vue:81]:
```
      [`btn-${color}`]: true,
```

Likewise here: string template + prop access

File: lib/components/base/Button.vue [hunk 4]
```diff
@@ -116,21 +93,12 @@ const accentedButton = computed(() =>
...
```
Comment (brawaru) [inline lib/components/base/Button.vue:101]:
```
      [`btn-${color}`]: true
```

String template + prop access

**Review by brawaru — CHANGES_REQUESTED** [review]

File: lib/components/base/Button.vue (hunk 2 above)
Comment (ToBinio) [reply lib/components/base/Button.vue]: done

File: lib/components/base/Button.vue (hunk 4 above)
Comment (ToBinio) [reply lib/components/base/Button.vue]: done

File: lib/components/base/Button.vue (hunk 3 above)
Comment (ToBinio) [reply lib/components/base/Button.vue]: done

File: lib/components/base/Button.vue (hunk 1 above)
Comment (ToBinio) [reply lib/components/base/Button.vue:31]: did decide to keep it based on the discussion on discord

**Pull request closed without merging**